packages/cozy-harvest-lib/src/components/KonnectorConfiguration/ConfigurationTab/BiContractActivationWindow.jsx

```diff
@@ -18,18 +19,48 @@ const BIContractActivationWindow = ({
   account,
   t,
   intentsApi,
-  innerAccountModalOverrides
+  innerAccountModalOverrides,
+  onAccountDeleted
 }) => {
   const [extraParams, setExtraParams] = useState(null)
   const [isWindowVisible, setWindowVisible] = useState(false)
+  const [isDeleteConnectionDialogVisible, setDeleteConnectionDialogVisible] =
+    useState(false)
   const [shouldRefreshContracts, setShouldRefreshContracts] = useState(false)
 
   const konnectorPolicy = findKonnectorPolicy(konnector)
   const client = useClient()
 
-  const onPopupClosed = () => {
+  /**
+   * Detects if a BI connection has been removed
+   *
+   * @param {String} finalLocation - url search param string from the final oauth location
+   * @returns {Boolean}
+   */
+  const isBIConnectionRemoved = finalLocation => {
+    const queryParams = new URLSearchParams(finalLocation)
+    return queryParams.get('connection_deleted') === 'true'
+  }
+
+  /**
+   * @typedef FinalOAuthRealtimeMessage
+   * @property {String} finalLocation - url search param string from the final oauth location
+   * @property {String|null} err - OAuth error message
+   * @property {String} oAuthStateKey - OAuth key
+   */
+
+  /**
+   *
+   * @param {String} key - OAuth key
+   * @param {FinalOAuthRealtimeMessage} oauthData
+   */
+  const onPopupClosed = (key, oauthData) => {
```

Comment:
oauthData is optional right? Then same message as before: https://github.com/cozy/cozy-libs/pull/1861/commits/c88a48d22001eb7ecde8ded5b0cfea26fac8eb74#r1007657923

Also, if `oauthData` is really optional, we should not call `isBiConnectionRemoved` in that case. Calling this method with `undefined` is not pertinent. 

```
if(oauthData && oauthData.finalLocation){
setDeleteConnectionDialogVisible(true)
}
...
```